webrender/src/debug_render.rs

```diff
@@ -398,6 +398,10 @@ impl DebugRenderer {
                     &self.tri_vertices,
                     VertexUsageHint::Dynamic,
                 );
+                #[cfg(not(feature = "gl"))]
+                {
+                    device.bind_pipeline();
```

Comment:
why isn't this done automatically inside `bind_program`?

Reply:
Because we need to know the blend state for bind the pipeline, and there are cases where the blend state is set after we bound a program or changed while a program is bound. A good example is drawing with `composite` where both can happen: https://github.com/szeged/webrender/blob/1dabf6a6560d52c0b6b2e0eca020c99c6a3db651/webrender/src/renderer.rs#L4389-L4427

Reply:
we should fix it then (both here and in upstream), so that the program is always bound after the blend, and enforce at run-time (e.g. having a flag in debug-only that tracks if the shader was bound after the last blend). There aren't many places where this is violated at the moment.

I'll take care of that work upstream.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1604664